easybuild/tools/modules.py

```diff
@@ -1469,7 +1475,17 @@ def prepend_module_path(self, path, set_mod_paths=True, priority=None):
         # Lmod pushes a path to the front on 'module use', no need for (costly) 'module unuse'
         modulepath = curr_module_paths()
         if not modulepath or os.path.realpath(modulepath[0]) != os.path.realpath(path):
+            # If no explicit priority is set, but priorities are already in use we need to use a high
+            # priority to make sure the path (very likely) ends up at the front
+            if priority is None and os.environ.get('__LMOD_Priority_MODULEPATH'):
+                priority = self.HIGH_PRIORITY
             self.use(path, priority=priority)
+            modulepath = curr_module_paths(clean=False)
+            path_idx = modulepath.index(path)
```

Comment:
This could trigger a `ValueError` if (somehow...) `path` is not in `$MODULEPATH`, so maybe wrap this in a `try`/`except` to avoid a nasty traceback?

It would be really weird to not have `path` at all in `modulepath` though, I must admit.

Reply:
It should be virtually impossible for this to happen as it should be either caught already by our tests or is an actual (and serious) bug in Lmod. So I'd say the "nasty traceback" is fine in this rare case so we don't need to litter our code :)